swarm/src/behaviour.rs

```diff
@@ -373,3 +387,18 @@ impl Default for DialPeerCondition {
         DialPeerCondition::Disconnected
     }
 }
+
+/// The options which connection handlers to disconnect.
+#[derive(Debug, Clone)]
+pub enum DisconnectPeerHandler {
```

Comment:
The docs mention the term `Options`, maybe `DisconnectOptions` is a better name for this?

I'd also agree to `DisconnectPeer` but `DisconnectPeerHandler` is a bit weird. In the end, we are disconnecting the actual peer, not the handler.

Reply:
I suppose someone would like to disconnect a specified connection by its `ConnectionId`. `DisconnectPeer` may confuse in this case.
> In the end, we are disconnecting the actual peer, not the handler.

I tried to follow the style of naming structures and enumerations that you follow :)
For example, [here](https://github.com/libp2p/rust-libp2p/blob/master/swarm/src/behaviour.rs#L338).
> The docs mention the term Options, maybe DisconnectOptions is a better name for this?

Reply:
> I tried to follow the style of naming structures and enumerations that you follow :)

I did see that and thought you would have been inspired by `NotifyHandler` :)

What about `CloseConnection::{One,All}`? That naming would be inline with [`ProtocolsHandlerEvent::Close`](https://github.com/libp2p/rust-libp2p/blob/e8fed53598696a45a26866408534cfa186b23d4a/swarm/src/protocols_handler.rs#L322) although that rename would affect the overall terminology used in PR.